src/System.IO.Compression.Brotli/src/System/IO/Compression/BrotliStream.cs

```diff
@@ -71,6 +85,7 @@ protected override void Dispose(bool disposing)
                 _stream = null;
                 _encoder.Dispose();
                 _decoder.Dispose();
+                ReturnBufferToPool();
```

Comment:
nit: Should we move all the finalizer logic to the helper?
```C#
_stream = null;
_encoder.Dispose();
_decoder.Dispose();
```

Reply:
Seems reasonable (renamed, of course).